### ISSUE TITLE
expectRevert should support panic codes

### ISSUE THREAD

**Issue by axic** [OpenZeppelin/openzeppelin-test-helpers#149]
Since right now the second argument is expected to be a string (which assumes the signature of `Error(string)`), I think the easiest way to extend this would be if a number is found, treat it s a panic code (with the signature of `Panic(uint256)`).

See https://docs.soliditylang.org/en/v0.8.0/control-structures.html#panic-via-assert-and-error-via-require for more information.

This would allow testing for specific abort reasons.

**Comment by abcoathup** [comment]
Hi @axic!  Thanks for the suggestion, it is really appreciated.

The project owner will review your suggestion as soon as they can. 

Please wait until we have discussed this idea before writing any code or submitting a Pull Request, so we can go through the design beforehand.  We don’t want you to waste your time!

**Comment by frangio** [comment]
Yes this would be great. I don't know how we would implement it though. We rely on the error message that Truffle emits. What error message does Truffle emit on a `Panic(uint256)` error?

**Comment by axic** [comment]
> What error message does Truffle emit on a Panic(uint256) error?

Hm, without looking, I hoped this would be based on examining the exact returndata. But it seems that ethereumjs/ganache does some processing.

[ethereumjs-vm](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/vm/lib/evm/eei.ts#L342-L350) returns the data unmodified. [ganache](https://github.com/trufflesuite/ganache-core/blob/develop/src/chains/ethereum/utils/src/errors/runtime-error.ts#L37-L55) looks up revert strings, but drops anything else.

Created an issue upstream: https://github.com/trufflesuite/ganache-core/issues/758

**Comment by frangio** [comment]
Thanks for looking into this. From a quick test Hardhat doesn't support Panic either, so I opened an issue there too. https://github.com/nomiclabs/hardhat/issues/1209
